torch/csrc/generic/methods/TensorMath.cwrap

```diff
@@ -119,8 +179,14 @@
 
 [[
   name: exp
-  defined_if: defined(TH_REAL_IS_FLOAT) || defined(TH_REAL_IS_DOUBLE) || CUDA_FLOAT || CUDA_HALF || CUDA_DOUBLE
-  with_stateless: True
+  types:
+    - floating_point
+  processors:
```

Comment:
Backend is probably a more forward-looking name as well. AMD backend sounds better than AMD processor.